config/traefik.yaml

```diff
@@ -0,0 +1,113 @@
+entrypoints:
+  # Defines a secure entrypoint using TLS encryption
+  websecure:
+    address: ":443"
+    http:
+      tls: true
+  # Defines an insecure entrypoint that redirects to the secure one.
+  web:
+    address: ":80"
+    http:
+      # Redirects HTTP to HTTPS
+      redirections:
+        entrypoint:
+          to: "websecure"
+          scheme: "https"
+
+api:
+  insecure: true
+  dashboard: true
+
+http:
+  services:
+    grist:
+      loadBalancer:
+        servers:
+          - url: 'http://grist:8484'
+    dex:
+      loadBalancer:
+        servers:
+          - url: 'http://dex:5556'
+    authelia:
+      loadBalancer:
+        servers:
+          - url: 'http://authelia:9091'
+
+  middlewares:
+    no-fwd:
+      headers:
+        customRequestHeaders:
+          'X-Forwarded-User': ""
+
+  routers:
+    route-grist-login:
+      rule: "PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`)"
+      service: grist
+      entryPoints:
+        - web
+
+    route-grist:
+      rule: "PathPrefix(`/`)"
+      priority: 1   # Set a lower priority than the other rules
+      service: grist
+      middlewares:
+        - no-fwd
+      entryPoints:
+        - web
+
+    route-dex:
+      rule: "PathPrefix(`/dex/`) || Path(`/dex`)"
+      service: dex
+      entryPoints:
+        - web
+
+    route-authelia:
+      rule: "PathPrefix(`/authelia/`) || Path(`/authelia`)"
+      service: authelia
+      entryPoints:
+        - web
+
+{{ $use_https := env "USE_HTTPS" }}
+{{if eq $use_https "true" }}
+    https-route-grist-login:
+      rule: 'Host(`{{ env "APP_HOST" }}`) && (PathPrefix(`/auth/login`) || PathPrefix(`/_oauth`))'
+      service: grist
+      entryPoints:
+        - websecure
+      tls: {{ env "TLS" }}
+
+    https-route-grist:
+      rule: 'Host(`{{ env "APP_HOST" }}`) && PathPrefix(`/`)'
+      priority: 1   # Set a lower priority than the other rules
+      service: grist
+      middlewares:
+        - no-fwd
```

Comment:
What does this no-fwd middleware do? Protect Grist from attempts at spoofing Forward Auth?

Reply:
There's an explanation by Dmitry from Omnibus when it was introduced: https://github.com/gristlabs/grist-omnibus/pull/10

Since we don't use forward-auth, this may be a vestige I can now remove. @dsagal is that correct?

Reply:
That is specifically for Forward Auth. In Omnibus, we set `GRIST_FORWARD_AUTH_HEADER` to `'X-Forwarded-User'`, which tells Grist to trust that header. So it's our responsibility within Omnibus to either set that header or omit it; just passing it through becomes dangerous. If not using forward auth (i.e. `GRIST_FORWARD_AUTH_HEADER` is NOT set), then no need for this middleware.

Reply:
Thanks for the explanation, I've removed it.